test/e2e/e2e_test.go

```diff
@@ -200,6 +202,90 @@ func TestE2EFlow(t *testing.T) {
 		VerifyIncreaseOfUserAccountCount(t, originalToolchainStatus, currentToolchainStatus, targetedJohnMur.Spec.UserAccounts[0].TargetCluster, 1)
 	})
 
+	t.Run("verify userAccount is not deleted if namespace is not deleted", func(t *testing.T) {
+		// given
+		johnMUR, err := hostAwait.WaitForMasterUserRecord(johnsmithName)
+		require.NoError(t, err)
+
+		fmt.Printf("\n >>>>>>>>>>>>>>>>>>>>>>>>>>>> MUR fetched>>>>>>>>>>>>>>>>")
+		// set deletion timestamp and add finalizer
+		deletionTS := metav1.Now()
+		johnMUR.DeletionTimestamp = &deletionTS
+		coputil.AddFinalizer(johnMUR, "finalizer.toolchain.dev.openshift.com")
+		if err := hostAwait.Client.Update(context.TODO(), johnMUR); err != nil {
+			fmt.Errorf("failed to set deletionTS and finalizer")
+		}
+		fmt.Printf("\n Deletion TS")
+		fmt.Printf(johnMUR.DeletionTimestamp.String())
+		fmt.Printf("\n finalizer")
+		fmt.Printf(johnMUR.Finalizers[0])
+
+		// when
+		//Create a pod with finalizer in user's namespace, which will block the deletion of namespace.
+		memberPod := corev1.Pod{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       "Pod",
+				APIVersion: "v1",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "test-useraccount-delete-1",
+				Namespace: "johnsmith-dev",
+				Finalizers: []string{
+					"kubernetes",
```

Comment:
you should add another finalizer here. `kubernetes` is the default finalizer IIRC, but you should add your own to prevent the pod deletion and make sure that the namespace is stuck in a `terminating` phase until your finalizer on the pod is removed

Reply:
the namespace is stuck in terminating state even with kubernetes finalizer